### ISSUE TITLE
Charachter offset

### ISSUE THREAD

**Issue by mortezakz** [mit-nlp/MITIE#190]
Hello, how do you get the character offset of a detected entity? I know you can get the number of the token, but what is the start / end character offset for the tokens in entities in the original string? 

Thanks.

**Comment by mortezakz** [comment]
Clarification, I am using Java.

**Comment by davisking** [comment]
Isn't the token's number the offset?

**Comment by mortezakz** [comment]
The token number just gives you .... the token number. For instance, in the following, the token number of the entity would be 2 and 3, but the character offset in the original string would be 7.

I like New York.

**Comment by davisking** [comment]
Right, but you should know where you got the token. You have to feed the tokens into MITIE. It’s not like it’s internally doing some hidden tokenization. 

So keep track of where you got the tokens from.

**Comment by davisking** [comment]
Or maybe you just don't see that the tokenizer is returning the offsets?  tokenizeWithOffsets() gives you the numbers.

**Comment by mortezakz** [comment]
Thank you very much.

Yes, I had not seen tokenizeWithOffsets(). I followed the Java example for your NER, and was looking only into the generated StringVector and not seeing the offsets. Didn't realize there was another class.

Where do you recommend I look for Java documentation or source? 



Also, FYI, I am using the 0.8 distribution on Maven. Currently writing a paper with a section on NER performance comparison of different tools on a test dataset, so please let me know if you have a more up-to-date version so that I report your best results.

Thanks again.

**Comment by davisking** [comment]
There isn't much to the java API.  You can see the binding here https://github.com/mit-nlp/MITIE/blob/master/mitielib/java/swig_api.h.  For detailed API docs for MITIE you can refer to the C API.  

MITIE hasn't changed in years, so any copy should be just as fast as any other.

**Comment by davisking** [comment]
No problem :)
